encrypt/encrypt.go

```diff
@@ -27,14 +36,288 @@ var ArgHelp = `
+var outDir = Args.String("outdir", "", "Output directory for encrypted files")
+
 // Main encryption function
-func Encrypt(args []string) {
+func Encrypt(args []string) error {
+
 	// Parse flags. There are no flags at the moment, but in case some are added
 	// we check for them.
 	Args.Parse(os.Args[1:])
 
 	// Args() returns the non-flag arguments, which we assume are filenames.
-	files := Args.Args()
+	// All filenames are read into a struct together with their output filenames
+	files := []encryptionFileSet{}
+	for _, filename := range Args.Args() {
+
+		// Set directory for the output file
+		outFilename := filename + ".c4gh"
+		if *outDir != "" {
+			_, basename := path.Split(filename)
+			outFilename = path.Join(*outDir, basename) + ".c4gh"
+		}
+
+		files = append(files, encryptionFileSet{filename, outFilename})
+	}
+
+	// Check that all the infiles exist, and all the outfiles don't
+	err := checkFiles(files)
+	if err != nil {
+		return err
+	}
+
+	// Read the public key to be used for encryption. The private key
+	// matching this public key will be able to decrypt the file.
+	publicKey, err := readPublicKey(*publicKeyFile)
+	if err != nil {
+		return err
+	}
+
+	// Generate a random private key to encrypt the data
+	privateKey, err := generatePrivateKey()
+	if err != nil {
+		return err
+	}
+
+	// Open all checksum files
+	ChecksumFileUnencMd5, err := os.OpenFile("checksum_unencrypted.md5",
+		os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer ChecksumFileUnencMd5.Close()
+
+	ChecksumFileUnencSha256, err := os.OpenFile("checksum_unencrypted.sha256",
+		os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer ChecksumFileUnencSha256.Close()
+
+	ChecksumFileEncMd5, err := os.OpenFile("checksum_encrypted.md5",
+		os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer ChecksumFileEncMd5.Close()
+
+	ChecksumFileEncSha256, err := os.OpenFile("checksum_encrypted.sha256",
+		os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer ChecksumFileEncSha256.Close()
+
+	// encrypt the input files
+	num_files := len(files)
+	for i, file := range files {
+		log.Infof("Encrypting file %v/%v: %s", i+1, num_files, file.unencrypted)
+
+		// encrypt the file
+		err = encrypt(file.unencrypted, file.encrypted, *publicKey, *privateKey)
+		if err != nil {
+			return err
+		}
+		// calculate hashes
+		hashes, err := calculateHashes(file)
+		if err != nil {
+			return err
+		}
+
+		// Write hashes
+		ChecksumFileUnencMd5.WriteString(
+			fmt.Sprintf("%s %s\n", hashes.unencryptedMd5, file.unencrypted))
+		ChecksumFileUnencSha256.WriteString(
+			fmt.Sprintf("%s %s\n", hashes.unencryptedSha256, file.unencrypted))
+		ChecksumFileEncMd5.WriteString(
+			fmt.Sprintf("%s %s\n", hashes.encryptedMd5, file.encrypted))
+		ChecksumFileEncSha256.WriteString(
+			fmt.Sprintf("%s %s\n", hashes.encryptedSha256, file.encrypted))
```

Comment:
```suggestion
		// Write hashes
		if _, err := ChecksumFileUnencMd5.WriteString(fmt.Sprintf("%s %s\n", hashes.unencryptedMd5, file.unencrypted)); err != nil {
			return err
		}

		if _, err := ChecksumFileUnencSha256.WriteString(fmt.Sprintf("%s %s\n", hashes.unencryptedSha256, file.unencrypted)); err != nil {
			return err
		}

		if _, err := ChecksumFileEncMd5.WriteString(fmt.Sprintf("%s %s\n", hashes.encryptedMd5, file.encrypted)); err != nil {
			return err
		}

		if _, err := ChecksumFileEncSha256.WriteString(fmt.Sprintf("%s %s\n", hashes.encryptedSha256, file.encrypted)); err != nil {
			return err
		}
```

Always remember to check for errors